externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/SparkSQLEngineListener.scala

```diff
@@ -61,9 +67,22 @@ class SparkSQLEngineListener(server: Serverable) extends SparkListener with Logg
            s"${ENGINE_DEREGISTER_EXCEPTION_MESSAGES.key}, deregistering the engine.")
        }
 
-       deregisterInfo.foreach { info =>
-         error(info, cause)
-         server.discoveryService.asInstanceOf[EngineServiceDiscovery].stop()
+       deregisterInfo.foreach { din =>
+         val currentTime = System.currentTimeMillis()
+         if (lastFailureTime == 0 || currentTime - lastFailureTime < deregisterExceptionTTL) {
+           jobFailureNum.incrementAndGet()
+         } else {
+           info("It has been more than one deregisterExceptionTTL" +
+             s" [${ENGINE_DEREGISTER_EXCEPTION_TTL} ms] since last failure, restart counting.")
```

Comment:
should we use the value here instead of the config entry?

Reply:
……sorry，my mistake